new-lamassu-admin/src/pages/UserManagement/UserManagement.js

```diff
@@ -0,0 +1,214 @@
+import { useQuery } from '@apollo/react-hooks'
+import { makeStyles, Box, Chip } from '@material-ui/core'
+import gql from 'graphql-tag'
+import * as R from 'ramda'
+import React, { useState, useContext } from 'react'
+
+import AppContext from 'src/AppContext'
+import { Link } from 'src/components/buttons'
+import { Switch } from 'src/components/inputs'
+import TitleSection from 'src/components/layout/TitleSection'
+import DataTable from 'src/components/tables/DataTable'
+
+import styles from './UserManagement.styles'
+import ChangeRoleModal from './modals/ChangeRoleModal'
+import CreateUserModal from './modals/CreateUserModal'
+import EnableUserModal from './modals/EnableUserModal'
+import Reset2FAModal from './modals/Reset2FAModal'
+import ResetPasswordModal from './modals/ResetPasswordModal'
+
+const useStyles = makeStyles(styles)
+
+const GET_USERS = gql`
+  query users {
+    users {
+      id
+      username
+      role
+      enabled
+      last_accessed
+      last_accessed_from
+      last_accessed_address
+    }
+  }
+`
+
+const Users = () => {
+  const classes = useStyles()
+  const { userData } = useContext(AppContext)
+
+  const { data: userResponse } = useQuery(GET_USERS)
+
+  const [showCreateUserModal, setShowCreateUserModal] = useState(false)
+  const toggleCreateUserModal = () =>
+    setShowCreateUserModal(!showCreateUserModal)
+
+  const [showResetPasswordModal, setShowResetPasswordModal] = useState(false)
+  const toggleResetPasswordModal = () =>
+    setShowResetPasswordModal(!showResetPasswordModal)
+
+  const [showReset2FAModal, setShowReset2FAModal] = useState(false)
+  const toggleReset2FAModal = () => setShowReset2FAModal(!showReset2FAModal)
+
+  const [showRoleModal, setShowRoleModal] = useState(false)
+  const toggleRoleModal = () => setShowRoleModal(!showRoleModal)
+
+  const [showEnableUserModal, setShowEnableUserModal] = useState(false)
+  const toggleEnableUserModal = () =>
+    setShowEnableUserModal(!showEnableUserModal)
```

Comment:
Even this can become simpler with a reducer. You would have 2 actions, 'close' and 'open'.

```js
switch (action) {
  case 'close': 
     return initialState
  case 'open':
     return { ...initialState, [type]: true }
}
```